aws/src/main/java/org/apache/iceberg/aws/lambda/rest/LambdaRESTInvoker.java

```diff
@@ -0,0 +1,167 @@
+      Object requestBody,
+      Class<T> responseType,
+      Map<String, String> inputHeaders,
+      Consumer<ErrorResponse> errorHandler,
+      Consumer<Map<String, String>> responseHeaders) {
+    validateInputPath(path);
+    URI uri = buildUri(path, queryParams);
+
+    Map<String, String> headers;
+    String entity = null;
+    if (requestBody instanceof Map) {
+      headers = requestHeaders(inputHeaders, ContentType.APPLICATION_FORM_URLENCODED.getMimeType());
+      entity = RESTUtil.encodeFormData((Map<?, ?>) requestBody);
+    } else if (requestBody != null) {
+      headers = requestHeaders(inputHeaders, ContentType.APPLICATION_JSON.getMimeType());
+      entity = super.toJsonString(requestBody);
+    } else {
+      headers = requestHeaders(inputHeaders, ContentType.APPLICATION_JSON.getMimeType());
+    }
+
+    LambdaRESTRequest request =
+        ImmutableLambdaRESTRequest.builder()
+            .method(method.name())
+            .uri(uri)
+            .headers(headers)
+            .entity(entity)
+            .build();
+
+    InvokeResponse lambdaResponse;
+    try {
+      lambdaResponse =
+          lambda.invoke(
+              InvokeRequest.builder()
+                  .functionName(functionArn)
+                  .payload(SdkBytes.fromInputStream(LambdaRESTRequestParser.toJsonStream(request)))
+                  .build());
+    } catch (AwsServiceException e) {
+      throw new RESTException(e, "Error occurred while processing %s request", method);
+    }
+
+    LambdaRESTResponse response =
+        LambdaRESTResponseParser.fromJsonStream(lambdaResponse.payload().asInputStream());
+    responseHeaders.accept(response.headers());
+
+    if (response.code() == HttpStatus.SC_NO_CONTENT
+        || (responseType == null && isSuccessful(response.code()))) {
+      return null;
+    }
+
+    if (!isSuccessful(response.code())) {
+      throwFailure(response.code(), response.entity(), response.reason(), errorHandler);
+    }
+
+    if (response.entity() == null) {
+      throw new RESTException(
+          "Invalid (null) response body for request (expected %s): method=%s, path=%s, status=%d",
+          responseType.getSimpleName(), method.name(), path, response.code());
+    }
+
+    return parseResponse(response.entity(), responseType, response.code());
+  }
+
+  @Override
+  public void close() throws IOException {
+    lambda.close();
+  }
+
+  @Override
+  public void initialize(
+      String baseUri,
+      ObjectMapper objectMapper,
+      Map<String, String> headers,
+      Map<String, String> properties) {
+    super.initialize(baseUri, objectMapper, headers, properties);
+    LambdaRESTInvokerProperties lambdaRESTInvokerProperties =
+        new LambdaRESTInvokerProperties(properties);
+    this.functionArn = lambdaRESTInvokerProperties.functionArn();
+    this.lambda = LambdaRESTInvokerAwsClientFactories.from(properties).lambda();
+    this.baseHeaders = headers;
+    // TODO: support sigv4 signer
```

Comment:
sure

Reply:
I realize this could just be passed as the same config to the Lambda, and then if needed the Lambda can inject the signer. I will let the people who implemented this to decide @danielcweeks